sympy/core/function.py

```diff
@@ -828,6 +828,10 @@ class AppliedUndef(Function):
 
     def __new__(cls, *args, **options):
         args = list(map(sympify, args))
+        u = [a.name for a in args if isinstance(a, UndefinedFunction)]
+        if u:
+            raise ValueError('invalid argument%s: %s' % (
+                's'*(len(u) > 1), ', '.join(u)))
```

Comment:
Can you improve this error message so that it is easier for a user to understand what they did wrong? Maybe just clarify why the argument is invalid e.g. "Invalid argument: expecting an expression, not UndefinedFunction". Also probably TypeError makes more sense than ValueError here.

Reply:
No problem, will do it right away.